src/app/dispatcher.ts

```diff
@@ -33,29 +41,22 @@ export class Dispatcher {
       try {
         return request.getOrFail(param)
       } catch (e) {
-        const msg =
-          'Unable to inject ' +
-          param +
-          ' into ' +
-          route.controller.name +
-          '.' +
-          route.function
-        throw Error(msg)
+        throw new MindlessError(
+          `Unable to inject ${param} into ${route.controller.name} ${route.function}`
+        )
       }
     }
 
+    let subjectController: Controller
     try {
-      let subjectController: Controller = container.resolve(route.controller)
-      let args = params.map(getArgToInject)
-
-      return await (subjectController as any)[route.function](...args)
+      subjectController = container.resolve(route.controller)
     } catch (e) {
-      let body = {
-        'Error Message': e.message,
-        'Mindless Message':
-          'Unable to resolve requested controller or method make sure your _routes are configured properly'
-      }
-      return new Response(500, body)
+      throw new MindlessError(
+        `Failed to resolve controller from container. Error Message: ${e.message}`
+      )
     }
+    const args = params.map(getArgToInject)
```

Comment:
put this inside `try` so that `subjectController` can be const?

Reply:
ya but i want the error from mapping params to bubble up so then i would have to 
```
try {
// ...
const args = params.map(getArgsToInject)
// ...
} catch (e) {
  if (e instance of MindlessError) {
    throw e
  }
  throw new MindlessError('')
}